[2024-06-18T16:38:27.052+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-18T16:38:27.096+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: nba_data_pipeline.upload_news_to_s3 scheduled__2024-06-17T00:00:00+00:00 [queued]>
[2024-06-18T16:38:27.106+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: nba_data_pipeline.upload_news_to_s3 scheduled__2024-06-17T00:00:00+00:00 [queued]>
[2024-06-18T16:38:27.106+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-06-18T16:38:27.127+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): upload_news_to_s3> on 2024-06-17 00:00:00+00:00
[2024-06-18T16:38:27.138+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'nba_data_pipeline', 'upload_news_to_s3', 'scheduled__2024-06-17T00:00:00+00:00', '--job-id', '93', '--raw', '--subdir', 'DAGS_FOLDER/nba_news_stats_etl.py', '--cfg-path', '/tmp/tmphgxu9hi3']
[2024-06-18T16:38:27.141+0000] {standard_task_runner.py:91} INFO - Job 93: Subtask upload_news_to_s3
[2024-06-18T16:38:27.141+0000] {logging_mixin.py:188} WARNING - /home/ubuntu/nba_ns_venv/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=4761) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-06-18T16:38:27.141+0000] {standard_task_runner.py:63} INFO - Started process 4829 to run task
[2024-06-18T16:38:27.246+0000] {task_command.py:426} INFO - Running <TaskInstance: nba_data_pipeline.upload_news_to_s3 scheduled__2024-06-17T00:00:00+00:00 [running]> on host ip-172-31-45-144.ec2.internal
[2024-06-18T16:38:27.415+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Airflow' AIRFLOW_CTX_DAG_ID='nba_data_pipeline' AIRFLOW_CTX_TASK_ID='upload_news_to_s3' AIRFLOW_CTX_EXECUTION_DATE='2024-06-17T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-17T00:00:00+00:00'
[2024-06-18T16:38:27.417+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-18T16:38:27.434+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-18T16:38:27.434+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/ubuntu/nba_ns_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/nba_ns_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/nba_ns_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/nba_ns_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/nba_ns_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/dags/nba_news_stats_etl.py", line 168, in upload_to_s3
    import boto3
ModuleNotFoundError: No module named 'boto3'
[2024-06-18T16:38:27.460+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=nba_data_pipeline, task_id=upload_news_to_s3, run_id=scheduled__2024-06-17T00:00:00+00:00, execution_date=20240617T000000, start_date=20240618T163827, end_date=20240618T163827
[2024-06-18T16:38:27.471+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 93 for task upload_news_to_s3 (No module named 'boto3'; 4829)
[2024-06-18T16:38:27.492+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-18T16:38:27.518+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
